source/addonHandler/__init__.py

```diff
@@ -181,26 +181,28 @@ def _getAvailableAddonsFromPath(path):
 		if p.endswith(DELETEDIR_SUFFIX): continue
 		addon_path = os.path.join(path, p)
 		if os.path.isdir(addon_path) and addon_path not in ('.', '..'):
-			log.debug("Loading add-on from %s", addon_path)
-			try:
-				a = Addon(addon_path)
-				name = a.manifest['name']
-				log.debug(
-					"Found add-on {name} - {a.version}."
-					" Requires API: {a.minimumNVDAVersion}."
-					" Last-tested API: {a.lastTestedNVDAVersion}".format(
-						name=name,
-						a=a
-					))
-				if a.isDisabled:
-					log.debug("Disabling add-on %s", name)
-				if not isAddonCompatible(a):
-					log.debugWarning("Add-on %s is considered incompatible", name)
-					_blockedAddons.add(a.name)
-				yield a
-			except:
-				log.error("Error loading Addon from path: %s", addon_path, exc_info=True)
-
+			if(len(os.listdir(addon_path))):

```

Comment:
In python the brackets are not necessary around the condition. Also, I suggest putting the error handling branch first so that the reader doesn't have to store it in their head through the longer "happy path". This requires inverting the condition logic.
```suggestion
			if not len(os.listdir(addon_path)):
```